homeassistant/components/overkiz/executor.py

```diff
@@ -76,9 +76,12 @@ async def async_execute_command(self, command_name: str, *args: Any) -> None:
         ):
             args = args + (0,)
 
+        if all(arg is None for arg in args):
+            args = ()
```

Comment:
Why pass `None` to begin with? `*args` also matches 0 positional args.

Reply:
That was my other thought. Based on feedback I refactored the AlarmControlPanel platform, however now it passes None if there are no arguments.

https://github.com/home-assistant/core/blob/91404041e08beab99b8cd1c48b372b617f8e6bcc/homeassistant/components/overkiz/alarm_control_panel.py#L267-L273

Would it be better to catch this in the alarm_control_panel implementation and make sure that we don't pass any arguments, when there are `None` args?

Reply:
Thanks @balloob. I still need to verify the fix, but this looks like the right direction. Now I understand your first comment as well, I was looking in the wrong direction :-).

Reply:
also, I am confused,

If you want to spread / unspread args, you need to do it on both sides:

https://github.com/home-assistant/core/blob/91404041e08beab99b8cd1c48b372b617f8e6bcc/homeassistant/components/overkiz/alarm_control_panel.py#L267-L273

You don't pass *args to it here, so it's a single argument that you then match with `*args` in the function definition. So your command now becomes [arg]. Which can be an embedded list. 

Also, instead of hardcoding empty lists in the entity description, to save a little memory, you could just do `args or []` when calling it . That way you still pass a list. 

Also, just drop the `*args` and make it `args`.

Reply:
```
>>> def print_args(*args):
...   print(args)
... 
>>> print_args(None)
(None,)
>>> print_args()
()
>>> print_args(*[1, 2])
(1, 2)
>>> print_args(*[])
()
```

Reply:
I agree that we need to refactor this piece. It is quite legacy and probably part of the old tahoma integration. I will try to refactor it for upcoming release. If I verify that the current fix works, would this be accepted for current release?

Reply:
Yeah that's fine.